packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -403,7 +403,8 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
       final GrowthDirection growthDirection = pivotParent.constraints.growthDirection;
       switch (applyGrowthDirectionToAxisDirection(axisDirection, growthDirection)) {
         case AxisDirection.up:
-          leadingScrollOffset = pivot.size.height - bounds.bottom;
+          final double offset = growthDirection == GrowthDirection.forward ? bounds.bottom : bounds.top;
```

Comment:
prefer a switch when dealing with an enum. That way if we ever add an enum value, the analyzer will tell us about every place we need to update.